app/recorder.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package app
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+
+	"github.com/mendersoftware/deviceconnect/store"
+)
+
+type Recorder struct {
+	sessionID string
+	store     store.DataStore
+	ctx       context.Context
+}
+
+const (
+	//It is used as a length of a memory region in bytes that is used to buffer
+	//the session recording. 4455 comes from the estimated typical terminal size in
+	//bytes (height=135 width=33) multiplied by 4 bytes of of terminal codes to get
+	//an estimate of a typical screen frame size in bytes.
+	RecorderBufferSize = 4 * 4455
```

Comment:
[Nitpick] I think it's more memory-wise to keep it page-aligned.
```suggestion
	RecorderBufferSize = 4 * 4096
```

Reply:
I *always* use n*1024 where n>=0 :) and this time I just did not.

Reply:
fixed.